libs/langchain/langchain/vectorstores/sqlitevss.py

```diff
@@ -0,0 +1,223 @@
+        )
+        self._connection.execute(
+            f"""
+                CREATE VIRTUAL TABLE IF NOT EXISTS vss_{self._table} USING vss0(
+                  text_embedding({self.get_dimensionality()})
+                );
+            """
+        )
+        self._connection.execute(
+            f"""
+                CREATE TRIGGER IF NOT EXISTS embed_text 
+                AFTER INSERT ON {self._table}
+                BEGIN
+                    INSERT INTO vss_{self._table}(rowid, text_embedding)
+                    VALUES (new.rowid, new.text_embedding) 
+                    ;
+                END;
+            """
+        )
+        self._connection.commit()
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Add more texts to the vectorstore index.
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+            kwargs: vectorstore specific parameters
+        """
+        max_id = self._connection.execute(
+            f"SELECT max(rowid) as rowid FROM {self._table}"
+        ).fetchone()["rowid"]
+        if max_id is None:  # no text added yet
+            max_id = 0
+
+        embeds = self._embedding.embed_documents(list(texts))
+        if not metadatas:
+            metadatas = [{} for _ in texts]
+        data_input = [
+            (text, json.dumps(metadata), json.dumps(embed))
+            for text, metadata, embed in zip(texts, metadatas, embeds)
+        ]
+        self._connection.executemany(
+            f"INSERT INTO {self._table}(text, metadata, text_embedding) "
+            f"VALUES (?,?,?)",
+            data_input,
+        )
+        self._connection.commit()
+        # pulling every ids we just inserted
+        results = self._connection.execute(
+            f"SELECT rowid FROM {self._table} WHERE rowid > {max_id}"
+        )
+        return [row["rowid"] for row in results]
+
+    def similarity_search_with_score_by_vector(
+        self, embedding: List[float], k: int = 4, **kwargs: Any
+    ) -> List[Tuple[Document, float]]:
+        sql_query = f"""
+            SELECT 
+                text,
+                metadata,
+                distance
+            FROM {self._table} e
+            INNER JOIN vss_{self._table} v on v.rowid = e.rowid  
+            WHERE vss_search(
+              v.text_embedding,
+              vss_search_params('{json.dumps(embedding)}', {k})
+            )
+        """
+        cursor = self._connection.cursor()
+        cursor.execute(sql_query)
+        results = cursor.fetchall()
+
+        documents = []
+        for row in results:
+            metadata = json.loads(row["metadata"]) or {}
+            doc = Document(page_content=row["text"], metadata=metadata)
+            score = self._euclidean_relevance_score_fn(row["distance"])
```

Comment:
i think we probably want to return raw distance here (the similarity_search_with_relevance_score method is where we'd want to normalize)

Reply:
OK, I'll adjust accordingly 👍